apps/remix-ide/src/app/files/workspaceFileProvider.js

```diff
@@ -74,9 +76,14 @@ class WorkspaceFileProvider extends FileProvider {
   }
 
   _normalizePath (path) {
-    if (!this.workspace) throw new Error('No workspace has been opened.')
+    if (!this.workspace) this.createWorkspace()
     return path.replace(this.workspacesPath + '/' + this.workspace + '/', '')
   }
+
+  createWorkspace (name) {
+    if (!name) name = 'default_workspace'
+    this.event.trigger('create_workspace', [name])
```

Comment:
name of the events are usually camel case I think `createWorkspace`

Reply:
ok